### ISSUE TITLE
Additional feature: LV2 plugin

### ISSUE THREAD

**Issue by Rezzonics** [jatinchowdhury18/KlonCentaur#28]
This is a great job, congratulations.
Would it be possible to have an LV2 plugin version of this pedal?

**Comment by dromer** [comment]
What about LV2 builds for windows and mac?

For MOD (arm64) we had to modify the toolchain, but have ChowCentaur working here: https://github.com/moddevices/mod-plugin-builder/compare/juce-tests
AnalogTapemodel is still in todo.

**Comment by jatinchowdhury18** [comment]
> What about LV2 builds for windows and mac?

Took a look at it this morning. There is a now an [`lv2` branch](https://github.com/jatinchowdhury18/KlonCentaur/tree/lv2), which contains a CMake build flag to enable LV2 builds on Windows/Mac (LV2 builds are still enabled by default on Linux). So to use this flag, you can run your CMake configuration step as:
```bash
cmake -Bbuild -G<My Generator> -DBUILD_LV2_TARGET=ON
```
That said, I've had some difficulty getting builds that I can test. On Windows, MSVC has trouble with the lv2_ttl_generator, since the [generator code](https://github.com/lv2-porting-project/JUCE/blob/lv2/extras/Build/lv2_ttl_generator/lv2_ttl_generator.c#L53) uses variable length arrays in a couple spots. Compiling with MINGW will probably work, but I haven't tried it yet.

On Mac, the CMake configuration step fails, since JUCE uses the "juceaide" tool to [generate a plist](https://github.com/lv2-porting-project/JUCE/blob/lv2/extras/Build/CMake/JUCEUtils.cmake#L1097), and juceaide does not recognize LV2 plugin targets, leading to a [false assertion](https://github.com/lv2-porting-project/JUCE/blob/lv2/extras/Build/juce_build_tools/utils/juce_ProjectType.h#L141).

Anyway, I'll keep experimenting, and see if I can get something to work!

Thanks,
Jatin

**Comment by dromer** [comment]
I really feel like juceaide is a huge curse. Nearly every project that uses juce is largely busy pulling in juce code and building juceaide ..

**Comment by jatinchowdhury18** [comment]
Yeah, it's definitely not the best, but I do feel that it's an improvement over the Projucer of old, both in terms of ease of use and build time. If the Projucer eventually goes away, then the time needed to pull JUCE from GitHub would go down considerably as well, since a large part of JUCE's memory footprint is taken up by the Visual Studio and Xcode projects needed for the Projucer.
